sailpoint.sgnl.yaml

```diff
@@ -0,0 +1,1575 @@
+# Sailpoint 2.0 Configuration YAML for DS2.0
+# Note: The externalId for each entity must match the endpoint for this resource i.e.
+# - External ID for an User entity representing a User resource (/Users) is "Users"
+# - External ID for a Group entity representing a Group resource (/Groups) is "Groups"
+
+displayName: "SailPoint"
+icon: PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHZpZXdCb3g9IjAgMCA2NCA2NCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTQwLjQxMTggNDIuMTM3M0gxMEwzNS4yMzUzIDEwTDQwLjQxMTggNDIuMTM3M1oiIGZpbGw9IiMwMDMzQTEiLz4KPHBhdGggZD0iTTQwLjQxMTggNDIuMTM3M0g1NEwzNS4yMzUzIDEwTDQwLjQxMTggNDIuMTM3M1oiIGZpbGw9IiNDQzI3QjAiLz4KPHBhdGggZD0iTTQwLjQxMTggNDIuMTM3M0gxMEw0Mi4zNTI5IDUzLjc4NDNMNDAuNDExOCA0Mi4xMzczWiIgZmlsbD0iIzAwNzFDRSIvPgo8cGF0aCBkPSJNNDAuNDExOCA0Mi4xMzczSDU0TDQyLjM1MjkgNTMuNzg0M0w0MC40MTE4IDQyLjEzNzNaIiBmaWxsPSIjRTE3RkQyIi8+Cjwvc3ZnPgo=
+description: "Sailpoint as a System of Record"
+address: "{{Input Required: Sailpoint URL}}"
+defaultSyncFrequency: HOURLY
+defaultSyncMinInterval: 1
+defaultApiCallFrequency: SECONDLY
+defaultApiCallMinInterval: 1
+type: "Sailpoint2.0-1.0.0"
+# Example Configs:
+# empty config: {} | b64: "e30="
+# config with filtering on Users: {"Users":{"queryParams":{"filter":"displayName eq \"SGNL\""}}} | b64: "eyJVc2VycyI6eyJxdWVyeVBhcmFtcyI6eyJmaWx0ZXIiOiJkaXNwbGF5TmFtZSBlcSBcIlNHTkxcIiJ9fQ=="
+# config with filter, ordering and sortBy on Groups {"Groups":{"queryParams":{"filter":"displayName eq \"SGNL\"","sortBy":"userName","ascending":true}}} | b64: "eyJHcm91cHMiOnsicXVlcnlQYXJhbXMiOnsiZmlsdGVyIjoiZGlzcGxheU5hbWUgZXEgXCJTR05MXCIiLCJzb3J0QnkiOiJ1c2VyTmFtZSIsImFzY2VuZGluZyI6dHJ1ZX19fQ=="
+adapterConfig: "e30=" # {}
+# This must match the list of supportedAuthMechanisms specified on the Adapter.
+auth:
+  - basic:
+      username: "{{Input Required: Username}}"
+      password: "{{Input Required: API Token}}"
+entities:
+  Account:
+    displayName: SailpointAccount
+    externalId: Accounts
+    description: Entity representing a Sailpoint Identity IQ Account.
+    syncFrequency: HOURLY
+    syncMinInterval: 1
+    apiCallFrequency: SECONDLY
+    apiCallMinInterval: 1
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: id
+        externalId: id
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: nativeIdentity
+        externalId: nativeIdentity
+        type: String
+      - name: displayName
+        externalId: displayName
+        type: String
+      - name: instance
+        externalId: instance
+        type: String
+      - name: uuid
```

Comment:
Should this be indexed (and potentially `uniqueId` instead of `id`)? How does this differ from `id`?

Reply:
The [schema](https://developer.sailpoint.com/iiq/api/get-accounts) does not tell much,

```
id: IdentityIQ id of the Account.
uuid: The UUID of the Account.
```

Example
```
"id": "c0b4568a4fe7458c434ee77d1fbt156b",
"uuid": "{f99999ff-f000-444b-b6ae-4443dd6cd6ed}",   
```

We can index it later if needed, based on the use case.